src/main/java/Storage/Storage.java

```diff
@@ -0,0 +1,238 @@
+        for (String str : taskInfo) {
+            if (!str.equals(")") && !str.equals("")) {
+                if (str.equals(separator)) {
+                    str = "/" + keyword;
+                }
+                output = output.concat(" " + str);
+            }
+        }
+        if (task.getIsDone()) {
+            output += " done";
+        }
+        return output + "\n";
+    }
+
+    /**
+     * Function detects if the possible Task created by the input is
+     * already presented in the current TaskList
+     * @param action is a string contains the command.
+     * @param curTaskList is the current TaskList object in the main Duke
+     * @return a boolean value shows if the action creates a duplicate task
+     */
+    public boolean isTaskDuplicate(String action, TaskList curTaskList) {
+        for (Task curTask : curTaskList.getTaskList()) {
+            if (formatTaskIntoCommands(curTask).equals(action)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Function able to read the input the judge which Task to use.
+     * @param action contains information for what task to be created.
+     * @return the task created.
+     * @throws Exception if format is invalid.
+     */
+    public Task createTask(String action) throws Exception{
+        // separate using the / as its the point of the date
+        String[] dataOfAction = action.split("/")[0].split(" ");
+        String type = dataOfAction[0];
+        //decide the correct header
+        switch (type) {
+            case ("Todo"): return new TodoTasks(action);
+
+            case ("Deadline"): return new DeadlineTask(action);
+
+            case ("Events") : return new EventsTask(action);
+
+            default: throw new Exception("Error type");
+        }
+    }
+}
+
+class TodoTasks extends Task{
+    /**
+     * constructor for Task.
+     *
+     * @param action contains all the information to build a TodoTask.
+     * @throws TodoException if the String action is invalid.
+     */
+    public TodoTasks(String action) throws TodoException {
+        super(action, "[T][] ", TaskTypes.Todo);
+        if (action.split(" ").length < 2){
+            throw new TodoException("OOPS!!! The description of a todo cannot be empty.");
+        }
+    }
+}
+
+class DeadlineTask extends Task{
+    /**
+     * constructor for Task.
+     *
+     * @param action containinf all the data needed for the creation of Deadline Task.
+     * @throws DeadlineException if the String action is invalid.
+     */
+    public DeadlineTask(String action) throws DeadlineException {
+        super(action, "[D][] ", TaskTypes.Deadline);
+        if (action.split("/").length < 2) {
+            throw new DeadlineException("incorrect format for deadline task");
+        }
+        if (action.split("/")[1].split(" ").length < 2) {
+            throw new DeadlineException("incorrect date format for Deadline task");
+        }
+    }
+}
+
```

Comment:
Should make these classes in separate java files rather than within another file.